dbms/src/Storages/Transaction/RegionBlockReader.cpp

```diff
@@ -37,18 +38,51 @@ RegionBlockReader::RegionBlockReader(DecodingStorageSchemaSnapshotConstPtr schem
     : schema_snapshot{std::move(schema_snapshot_)}
 {}
 
+
 bool RegionBlockReader::read(Block & block, const RegionDataReadInfoList & data_list, bool force_decode)
 {
-    switch (schema_snapshot->pk_type)
+    try
+    {
+        switch (schema_snapshot->pk_type)
+        {
+        case TMTPKType::INT64:
+            return readImpl<TMTPKType::INT64>(block, data_list, force_decode);
+        case TMTPKType::UINT64:
+            return readImpl<TMTPKType::UINT64>(block, data_list, force_decode);
+        case TMTPKType::STRING:
+            return readImpl<TMTPKType::STRING>(block, data_list, force_decode);
+        default:
+            return readImpl<TMTPKType::UNSPECIFIED>(block, data_list, force_decode);
+        }
+    }
+    catch (DB::Exception & exception)
```

Comment:
```suggestion
    catch (DB::Exception & e)
```
Better avoid using the fullname `exception`, it could be confusing with `std::exception`. Usually we use a short name `e` or `exc`